### PR TITLE
More AOT serialization fixes

### PR DESCRIPTION
**Pull request by morganbr**
More fixes for UAP AOT BinaryFormatter tests: 
* Making Cookie and CookieContainer serialize correctly (I'm not sure why this wasn't needed for CoreCLR)
* More rd.xml
* Avoid private reflection in the test Comparer comparison
* Minor tweaks to the byreflike test

### CODE REVIEW

**Comment by morganbr** [comment]
CC @dotnet/corert-contrib

File: src/System.Net.Primitives/src/System/Net/Cookie.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 namespace System.Net
 {
-    internal enum CookieVariant
+    public enum CookieVariant
```
Comment (danmosemsft) [inline src/System.Net.Primitives/src/System/Net/Cookie.cs]: Is this the change that you believe should have broken serializatoin for CoreCLR as well? Could you please check why the tst is apparently passing without it (ie. we have a test)?

Comment (ViktorHofer) [inline src/System.Net.Primitives/src/System/Net/Cookie.cs]: Serialization of Cookie/CookieContainer is/was broken between different frameworks and I was looking into it. So our tests in fact failed correctly and you were just faster finding it

File: src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs [hunk 2]
```diff
@@ -100,12 +100,22 @@ public static bool IsEqual(this DataTable @this, DataTable other)
 
         public static bool IsEqual(this Comparer @this, Comparer other)
```
Comment (ViktorHofer) [inline src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs]: All the equality checks should also be refined in future to test the whole object. It's on my list.

**Pull request merged**